### DISCUSSION TITLE
Chapel project meeting: 2026-04-07 (agenda items, meeting link, notes)

### DISCUSSION THREAD

**Discussion by github-actions** [chapel-lang/chapel#28621] [Weekly meeting topics]
This discussion is for the weekly Chapel project meeting.

**Date/Time: 2026-04-07 at 10am PT**
**Join the meeting using [this Zoom link](https://zoom-lfx.platform.linuxfoundation.org/meeting/97154927840?password=96736b6d-0900-4b20-81ba-ff6bf48668af)**
See the [Chapel community calendar](https://chapel-lang.org/calendar/) for additional details

---

Prior to the meeting, propose any agenda items using comments below.  After the meeting, rough notes will be added in comments.

Typically, these meetings consist of:
* **News:** Quick updates or announcements to keep people in the loop on key efforts or topics
* **Demos:** Have something cool to show off?  Do it in a ~5-minute demo
* **Short Topics:** Have something that needs a quick discussion, description, or straw poll?  Propose a ~5-minute discussion topic
* **Longer Topics:** If you have a topic needing more time, propose it with an estimated duration and we'll try to fit it in.

If you'd like to propose an item in any of these categories, please reply to this discussion.

In addition to the above sections, we have some standing agenda items, such as:
* **Triage:** What is the state of testing and are there issues that need more attention?
* **User Issues:** What new user issues have come up in the past week, and which should we prioritize?
* **Q&A:** Have something you want to ask, but not on the clock?  We'll open the floor to Q&A as the official meeting wraps up.

**Comment by jabraham17** [comment]
I'd like to have a discussion about https://github.com/chapel-lang/chapel/issues/28643

**Comment by jabraham17** [comment]
I have a demo of two new chpldoc features

**Comment by DanilaFe** [comment]
I have (up to) two demos:

* some progress on and applications of torch bindings for Chapel
* `MisleadingIndentation` + `IncorrectIndentation` improvements

**Comment by lydia-duncan** [comment]
Here's today's user issues slides

<img width="956" height="536" alt="Screenshot 2026-04-07 111033" src="https://github.com/user-attachments/assets/59de9d00-bd95-4c32-994b-7493060b643e" />

**Comment by bradcray** [comment]
Agenda / Notes
==========

News/Announcements
------------------

* Thursday deep-dive slot: no plans for this week at present
  - to propose a topic: https://github.com/chapel-lang/chapel/discussions/27247


Triage Status
-------------
* Brad: end-of-rotation update (Jade's up next)

  - code issues:
    - union-oriented failures: linux64, --fast, c-backend, asan, blog (Jade)
    - playground no longer merging cleanly with 'main' (DavidL)
    - sporadic ex-ofi failure for raCommCheckLCG (Brad)

  - use-case issues:
     - Arkouda sporadic failures (Engin->unowned)
       - seem to be due to real nondeterministic failures
       - caused by series of PRs that were merged quickly in late-Feb/early Mar
       - test_dataframe test is failing
       - Engin also thinks repeatedly using port 5555 is causing problems
         - could potentially be fixed by iterating through port numbers
     - v2: trying to skip failing test(s) for now if easy as hoped (Jade)
     - CHAMPS: trying to disable optional module to see if it fixes (Jade)

  - systems-ish issues (likely) within our control:
    - Docker: passed once last week, then failing w/ "failed to push" (Anna)
      - also still seemingly getting stuck sometimes
    - Packages: Colima is down and needs to be restarted (Brad)

  - systems-ish issues we could potentially be more resilient to (?):
    - apptainer: curl failure in 1 of 50 configs last night


User Issues
-----------
* Lydia: weekly report
  - 3 resolved user issues:
    - active field of union (Jade)
    - tuple vs. multi-var declaration confusion (self-resolved, user mistake)
       - Daniel: inclined to help users with this case, maybe under new issue?
    - compiler fails to resolve cross-module promoted call (Daniel)


Demos
-----
* Daniel: torch bindings for Chapel
  - have claude load pre-trained image module using torch
    - draws bounding boxes around objects in image
  - uses Jade's image module to load up the jpg and detect some people + a tie
  - code: load image, resize image, tensor ops, forward inference pass
  - model creates 3D tensor of various classifications and etc.
  - most routines are hand-wrapped external proc declarations
    - much of the it being auto-generated
  - wanted to know: can I run this in parallel?
    - did an experiment turning inference into a coforall loop
    - memory & CPU saturation contention seems to limit speedup
  - tried doing an HPO -- use each thread to train a different model
    - see which option leads to the best model
    - forall loop over number of model configurations
    - returns top 5 models, some of which have slightly better accuracy
      - 98% vs. 95%
      - the case it can't get is one that's pretty ambiguous between 5 and 6
  - have you tried distributed?
    - not yet
  - side discussion about how to get "I'm not sure" results out of the system
  - did parallelism on MNIST yield speedup?
    - yes, some... maybe 2-3x on 8 cores?
  - note that going to distributed memory could reduce contention for memory
    and/or CPU that could be limiting speedup

* Jade: chpldoc features
  - single-module docs are not particularly interesting or useful
    - front page is sort of boilerplate no-op
  - normal 'usage' stuff feels pedantic/repetitive
  - as user, may want to get rid of front page
    - write custom _index file that includes the module
    - but that gives warnings about duplicate inclusions
  - new attribute says "I'll manually include this"
    - nicely includes it in index, no errors, all works nicely
  - also new attribute to suppress standard "usage" header
  - feels like it leads to much more readable documentation
  - general sentiment seems to be to document these features
  - Q about whether a single-module page could support concise docs
    - rather than taking the (few) steps Jade had to
    - yes, just a different theme than Jade's been focusing on ("control")
    
* Daniel: linter indentation improvements
  - ran out of time for this, so pushing it to next week
  

Discussion Topics
-----------------
* Jade: re-assignment of dead, non-nilable, owned variables
  - spec shows an example that transfers an owned class
    - seems to pre-date adoption of non-nilable classes, and is not tested
    - so, gives impression that a dead non-nilable can be reassigned
    - Jade/Daniel would like that as a behavior
  - general agreement that the spec should be updated
    - ideally showing behavior for nilable and non-nilable cases
      - no need to show assignment both directions for nilable (maybe both?)
  - Brad: doesn't read spec as implying dead can be re-assigned
    - dates from when all classes were nilable, so made sense then
  - Brad: also reasonably confident this is behaving as intended
    - not to say the proposed feature is a bad one
    - but not obviously a bug fix to me
    - would be interesting to get Michael's take
    - (post-meeting: found connections to copy elision)
      - defined in terms of "last mention of identifier" currently

* thoughts about Zoom this week?
  - didn't get to this during meeting
  - (offline, HPE team members seemed amenable to Zoom-only)


Post-meeting
============


Office Hours / Q&A
------------------


TODOs / Open Issues
-------------------
- [x] cancel Thursday meeting if no plans
  - did this, but late
